### PR TITLE
[Google Blockly] Fix block highlighting

### PR DESCRIPTION
**Pull request by ajpal**
There was a regression where the currently-executing block was no longer being highlighted.
The reason was that Google Blockly block IDs are 20-character UIDs, not numbers, so this wasn't working https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/StudioApp.js#L1622

For the work I was doing on UI tests, I wanted to change the IDs to be numbers anyways, but when I did so, I noticed that the block highlight now looks like this:
Oct-28-2020 12-49-30

My assumption is that we don't want this, so I updated the corresponding path_object code to match our existing highlight behavior:
Oct-28-2020 12-51-04

### CODE REVIEW

File: apps/src/sites/studio/pages/googleBlocklyWrapper.js [hunk 1]
```diff
@@ -21,6 +21,15 @@ import CdoWorkspaceSvg from '@cdo/apps/blocklyAddons/cdoWorkspaceSvg';
 const BlocklyWrapper = function(blocklyInstance) {
   this.version = BlocklyVersion.GOOGLE;
   this.blockly_ = blocklyInstance;
+
+  /**
+   * If an id is not specified, the Block constructor will randomly generate a 20-character
+   * uid string. In UI tests, we want to be able to reference blocks by their id, which won't
```
Comment (breville) [inline apps/src/sites/studio/pages/googleBlocklyWrapper.js]: It sounds like this is no longer just for UI tests but also for runtime behaviour for block highlighting?

File: apps/src/blocklyAddons/cdoBlockSvg.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import BlockSvgUnused from './blockSvgUnused';
 
 export default class BlockSvg extends GoogleBlockly.BlockSvg {
   constructor(workspace, prototypeName, opt_id) {
-    super(workspace, prototypeName, opt_id);
+    super(workspace, prototypeName, ++Blockly.uidCounter_); // Use counter instead of randomly generated IDs
```
Comment (breville) [inline apps/src/blocklyAddons/cdoBlockSvg.js:6]: Unlikely, but is there any risk of overflow?

File: apps/src/sites/studio/pages/googleBlocklyWrapper.js [hunk 3]
```diff
@@ -21,6 +21,15 @@ import CdoWorkspaceSvg from '@cdo/apps/blocklyAddons/cdoWorkspaceSvg';
...
+   * work if the ids are randomly generated. To get around this problem, we can just use a
+   * global counter as the block ids.
+   */
+  this.uidCounter_ = 0;
```
Comment (breville) [inline apps/src/sites/studio/pages/googleBlocklyWrapper.js:31]: Does this only get initialized once when we load a level?

File: apps/src/sites/studio/pages/googleBlocklyWrapper.js (hunk 1 above)
Comment (ajpal) [reply apps/src/sites/studio/pages/googleBlocklyWrapper.js]: Well I could also just update the regex in StudioApp if we wanted.. The only place that has a firm requirement for these IDs to be predictable is the UI tests. But we do reference blocks by their id throughout runtime code as well as tests.

File: apps/src/blocklyAddons/cdoBlockSvg.js (hunk 2 above)
Comment (ajpal) [reply apps/src/blocklyAddons/cdoBlockSvg.js:6]: hmm, I guess it's possible? It's the same setup we've used for years with CDO Blockly (https://github.com/code-dot-org/blockly/blob/main/core/ui/block.js#L47) so I'm sort of inclined to just stick with it this way

File: apps/src/sites/studio/pages/googleBlocklyWrapper.js (hunk 3 above)
Comment (ajpal) [reply apps/src/sites/studio/pages/googleBlocklyWrapper.js:31]: Yeah, it's set when we initialize Blockly, which happens on page load

**Review by breville — APPROVED** [review]

**Pull request merged**